### ISSUE TITLE
fractional window length error with boxcar

### ISSUE THREAD

**Issue by bmcfee** [librosa/librosa#449]
From the [travis logs](https://api.travis-ci.org/jobs/177711565/log.txt?deansi=true):
```
test_filters.test__window(16.25, <function boxcar at 0x7fd5e51cae18>) ... /home/travis/env/miniconda3.5/envs/test-environment/lib/python3.5/site-packages/numpy/core/numeric.py:190: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  a = empty(shape, dtype, order)
ok
```

**Comment by bmcfee** [comment]
Digging into the [scipy source](https://github.com/scipy/scipy/blob/master/scipy/signal/windows.py) it looks like floating window lengths will break everything soon.  [Relvant code](https://github.com/scipy/scipy/blob/master/scipy/signal/windows.py#L16-L20):
```python
def _len_guards(M):
    """Handle small or incorrect window lengths"""
    if int(M) != M or M < 0:
        raise ValueError('Window length M must be a non-negative integer')
    return M <= 1
```
This function is called by all scipy window functions.

We have two options to work around this:

1. Reimplement all window functions without the length guard
2. Modify `float_window` to call with the floor of `n`.

I'm leaning toward option 2.  It will break some backwards compatibility, and introduce a small amount of error to the CQT basis construction, but I don't think it's worth adding a relatively large chunk of near-duplicate code.

What do others think?  @ebattenberg @dpwe @lostanlen

**Comment by bmcfee** [comment]
PS: this is implemented and ready to go in #452 , if nobody objects.

**Comment by bmcfee** [comment]
Nobody objected, so this one is closed.
